services/parser-database/connector.py

```diff
@@ -0,0 +1,70 @@
+"""conector.py - One stop connector for external
+services/databases"""
+import os
+import requests  # pylint: disable=import-error
+
+
+articles_in_memory = {}
+keywords_in_memory = {}
+
+
+def get_documents_to_parse():
+    # When database is integrated, this will go away
+    mty_document = {
+        "hash":
+        "afafbfbdce8c40924edae00f6ce54f0c639ce42a2" +
+        "c0fbbfa6ab82ea6925827c51",
+        "jurisdiction":
+        "Monterrey",
+        "url":
+        "http://www.guadalupe.gob.mx/wp-content/up" +
+        "loads/2019/09/Nuevo-Reglamento-Homologado-1.pdf",
+    }
+    document_list = []
+    document_list.append(mty_document)
+    return document_list
+
+
+def get_keywords(text):
+    """Get keywords that relate to this article (Javier's service)
+
+    Args:
+        text (sting): text to extract keywords from
+
+    Returns:
+        [list]: list of extracted keywords
+    """
+
+    extracted_keywords = []
+    request = {'text': text}
+    nlp_output = requests.post(os.getenv("KEYWORDS_SERVICE"), json=request)
+    # nlp_output = keywordmock.get_keywords(text)
+    json_output = nlp_output.json()
+    for keyword in json_output["tokens"]:
+        extracted_keywords.append(keyword["lemma"])
+    return extracted_keywords
+
+
+def get_article_by_number(art_num):
+    if art_num in articles_in_memory:
```

Comment:
I am not really sure how does the `in` keyword works in Python, but it seems that it traverses over all object attributes to find the article.

Check this [link](https://stackoverflow.com/a/610893/11672866) out. They use `hasattr()` method to find the attribute. I don't know if it's faster, but you should look at it.

Reply:
Avg: O(1) https://stackoverflow.com/questions/13884177/complexity-of-in-operator-in-python